ls1intum/Artemis
# `General`: Hide email addresses on default

### Checklist
#### General


#### Server
#### Client
#### Changes affecting Programming Exercises

### Motivation and Context


Fixes #4786 

### Description

I applied the same principle that already exists for the registration number to the email address.
After that I updated the client and went through every reference of the new `visibleEmail` and set the email to visible in the server resource.

**For now I added TODOs at places where I need to start a discussion whether to keep the email address there or not.**

### Steps for Testing

Ideas:
- Check discussion tab to verify that the original issue is resolved
- Check that user registration works
- Check that user creation and update works
- check that user see their own email address
- check that instructors still can edit the exam time of a student properly
- check that the course score export still works

### Review Progress





#### Code Review
#### Manual Tests
